### PR TITLE
Continuous Integration and Deployment

### PR DESCRIPTION
**Pull request by tonykimathi**
**What does this PR do?
This pull request adds multiple features to the shopping list app, such as continuous integration testing and test coverage.

### CODE REVIEW

**Review by michaelkagera2 — COMMENTED** [review]
From last review there is integration of flask. The requirements.txt is also there which covers all the requirements that are needed to run the app. I'm new to some of the modules so I can'tell really give feedback on them i.e werzeug. I can also see that you are using TDD. Changes that may help are adding a git ignore file. Other than that, everything seems good. Nice job. Looking forward to the whole app.

File: app/models.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+
+class User(object):
+    def __init__(self, name, email, password):
+        self.name = name
+        self.email = email
+        self.__password = password
+
+@property
+def password():
```
Comment (kkipyegon) [inline app/models.py:11]: why this function ?? it should be an instance method of user function

File: app/models.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    raise AttributeError("password is not a readable attribute.")
+
+@password.setter
```
Comment (kkipyegon) [inline app/models.py:15]: why this function ?? it should be an instance method of user function

File: app/models.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+def password(self, password):
+
+    self.password_hash = generate_password_hash(password)
+
+
+def verify_password(self, password):
```
Comment (kkipyegon) [inline app/models.py:21]: why this function ?? it should be an instance method of user function

Comment (tonykimathi) [reply app/models.py:21]: So should the function be under the User class as a method?

Comment (tonykimathi) [reply app/models.py:21]: Noted

**Comment by justmesam** [comment]
Remove the pycache and the coverage from the repo.
fix the failing build on codeclimate before you merge.
Other than that you project structure is on point and your commits are okey.

**Pull request merged**